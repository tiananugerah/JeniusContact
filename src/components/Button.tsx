import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';

interface ButtonProps {
  title: string;
  onPress: () => void;
}

const Button: React.FC<ButtonProps> = ({title, onPress}) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#007aff',
    paddingVertical: 12,
    paddingHorizontal: 24,
    borderRadius: 8,
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

export default Button;
