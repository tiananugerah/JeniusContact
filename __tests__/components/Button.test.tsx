// __tests__/Button.test.tsx

import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Button from '../../src/components/Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    const {getByText} = render(<Button title="Press me" onPress={() => {}} />);
    const buttonElement = getByText('Press me');
    expect(buttonElement).toBeTruthy();
  });

  it('calls onPress callback', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <Button title="Press me" onPress={onPressMock} />,
    );
    const buttonElement = getByText('Press me');
    fireEvent.press(buttonElement);
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const {toJSON} = render(<Button title="Press me" onPress={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
