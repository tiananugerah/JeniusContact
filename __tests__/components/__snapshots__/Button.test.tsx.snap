// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#007aff",
      "borderRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Press me
  </Text>
</View>
`;
